ably.ts.d

```diff
@@ -0,0 +1,400 @@
+
+  update: (Data: any, callback?: standardCallback) => void;
+
+  leave: (Data: any, callback?: standardCallback) => void;
+
+  enterClient: (clientId: String, Data: any, callback?: standardCallback) => void;
+
+  updateClient: (clientId: String, Data: any, callback?: standardCallback) => void;
+
+  leaveClient: (clientId: String, Data: any, callback?: standardCallback) => void;
+}
+
+export declare class Channel {
+  name: String;
+  presence: Presence;
+  history: (ParamsOrCallback?: RestPresenceHistoryParams | PaginatedResultCallback, callback?: PaginatedResultCallback ) => void;
+  publish: (messages: any, MessageDataOrCallback?: any | errorCallback, callback?: errorCallback) => void;
+}
+
+export declare class RealtimeChannel extends EventEmitter {
+  name: String;
+  errorReason: ErrorInfo;
+  state: ChannelState;
+  presence: RealtimePresence;
+
+  attach: (callback?: standardCallback) => void;
+
+  detach:(callback?: standardCallback) => void;
+
+  history: (ParamsOrCallback?: RealtimePresenceHistoryParams | PaginatedResultCallback, callback?: PaginatedResultCallback) => void;
+
+  subscribe: (eventOrCallback: String | messageCallback, listener?: messageCallback) => void;
+
+  unsubscribe: (eventOrCallback?: String | messageCallback, listener?: messageCallback) => void;
+
+  publish: (messages: any, MessageDataOrCallback?: any | errorCallback, callback?: errorCallback) => void;
+
+}
+
+export declare class Channels {
+  get: (name: String, channelOptions?: ChannelOptions) => RealtimeChannel;
+
+  release: (name: String) => void;
+}
+
+export declare class Message {
+  constructor();
+
+  fromEncoded: (JsonObject: String, channelOptions: ChannelOptions) => Message;
+
+  fromEncodedArray: (JsonArray: String, channelOptions: ChannelOptions) => Array<Message>;
+
+  clientId: String;
+  connectionId: String;
+  data: any;
+  encoding: String;
+  extras: String;
+  id: String;
+  name: String;
+  timestamp: number; // Time
+}
+
+export declare class PresenceMessage {
+  fromEncoded: (JsonObject: any, channelOptions?: ChannelOptions) => PresenceMessage;
+
+  fromEncodedArray: (JsonArray: Array<any>, channelOptions?: ChannelOptions) => Array<PresenceMessage>;
+
+  action: PresenceAction;
+  clientId: String;
+  connectionId: String;
+  data: any;
+  encoding: String;
+  id: String;
+  timestamp: number; // Time
+}
+
+export declare class Rest {
+  constructor(options: String | ClientOptions);
+  auth: Auth;
+  channels: Channels;
+  request: (method: string, path: string, parmas?: any, body?: Array<any> | any, headers?: any, callback?: (Error: ErrorInfo, results: HttpPaginatedResponse) => void) => void;
+  stats: (ParamsOrCallback?: any, callback?: PaginatedResultCallback) => void;
+  time: (paramsOrCallback?: any, callback?: (Error: ErrorInfo, time: number ) => void) => void; // time
+}
+
+export declare class Realtime {
+  constructor(options: String | ClientOptions);
+
+  auth: Auth;
+  channels: Channels;
```

Comment:
This should reference `Channels<RealtimeChannel>` as it does not return a REST `Channel`